### PR TITLE
Documentation for the resolution of Arnold's equation

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: documentation/Celledoni.tex [hunk 1]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\[
+\begin{dcases}
+p_2 &= \frac{p_1 m_2 + p_0 m_3}{G + m_1} \\
+p_3 &= \frac{p_1 m_3 - p_0 m_2}{G + m_1} \\
+p_0^2 + p_1^2 &= \frac{G + m_1}{2 G}
+\end{dcases}
+\]
+and, for the angle $\gy\of{t}$ of the rotation $\mathscr Y_t$ around $\VectorSymbol{e_1}$:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T + G m_1 / I_1}{G + m_1} + 4 G \frac{p_1 \TimeDerivative{p_0} - p_0 \TimeDerivative{p_1}}{G + m_1}
```
Comment (eggrobin) [inline documentation/Celledoni.tex:516]: rad<sup>2</sup>

File: documentation/Celledoni.tex [hunk 2]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\begin{dcases}
+p_0 &= \sqrt\frac{1 + m_1/G}{2} \\
+p_1 &= 0 \\
+p_2 &= \frac{m_3}{\sqrt{2 G\pa{G + m_1}}} \\
+p_3 &= \frac{-m_2}{\sqrt{2 G\pa{G + m_1}}}
+\end{dcases}
+\]
+and the angle to:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T + G m_1 / I_1}{G + m_1} = \frac{G^2 - \gD_1 + G m_1}{I_1\pa{G + m_1}} = \frac{G}{I_1} - \frac{\gD_1/I_1}{G + m_1} =
```
Comment (eggrobin) [inline documentation/Celledoni.tex:529]: rad<sup>2</sup>

File: documentation/Celledoni.tex [hunk 3]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\frac{G}{I_1} - \frac{\gD_1}{G I_1}\frac{1}{1 + m_1/G}
+\]
+Using equation (\ref{integraldn}) with $a = B_{13}/G$ and simplifying the various coefficients we obtain:
+\[
+\gy\of{t} = \frac{G}{I_1}t + \frac{G I_{13}}{\gl I_1 I_3}
+\EllipticPi\of{\JacobiAmplitude\of{\gl t - \gn}, \frac{I_1 I_{32}}{I_3 I_{12}}, k} -
+\InverseTrigonometricTangent\of{\sqrt\frac{I_2 I_{31}}{I_3 I_{21}} \JacobiSC\of{\gl t - \gn, k}}
```
Comment (eggrobin) [inline documentation/Celledoni.tex:536]: s/𝑘/𝑘<sup>-1</sup>/

File: documentation/Celledoni.tex [hunk 4]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+and the angle to:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T + G m_1 / I_1}{G + m_1} = \frac{G^2 - \gD_1 + G m_1}{I_1\pa{G + m_1}} = \frac{G}{I_1} - \frac{\gD_1/I_1}{G + m_1} =
+\frac{G}{I_1} - \frac{\gD_1}{G I_1}\frac{1}{1 + m_1/G}
+\]
+Using equation (\ref{integraldn}) with $a = B_{13}/G$ and simplifying the various coefficients we obtain:
+\[
+\gy\of{t} = \frac{G}{I_1}t + \frac{G I_{13}}{\gl I_1 I_3}
+\EllipticPi\of{\JacobiAmplitude\of{\gl t - \gn}, \frac{I_1 I_{32}}{I_3 I_{12}}, k} -
```
Comment (eggrobin) [inline documentation/Celledoni.tex:535]: s/𝑘/𝑘<sup>-1</sup>/

File: documentation/Celledoni.tex [hunk 5]
```diff
@@ -108,7 +110,27 @@ \section*{Solution of Euler's equation}
 value of $T$ .\label{figGT}}
 \end{figure}
 
-In the rest of this section, we derive (corrected) formul{\ae} for the three cases described above.
+In the rest of this section, we describe our notation and derive (corrected) formul{\ae} for the three cases described above.
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:113]: Done.

File: documentation/bibliography.bib [hunk 6]
```diff
@@ -681,6 +692,24 @@ @article{Yoshida1990
...
 }
 
+@article{ZhaoJiHuangHuHouLiIp2002,
+  author       = {Zhao, Yuhui and Ji, Jianghui and Huang, Jiangchuan and Hu, Shoucun and Hou, Xiyun and Li, Yuan and Ip, Wing-Huen},
+  date         = {2015-05},
+  doi          = {10.1093/mnras/stv792},
+  journaltitle = {Monthly Notices of the Royal Astronomical Society},
+  number       = {4},
+  pages        = {3620--3632},
+  title        = {Orientation and rotational parameters of asteroid 4179 Toutatis: new insights from Chang′e-2's close flyby},
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/bibliography.bib:702]: Done, but note that the `PRIME` comes from the article title.

File: documentation/Celledoni.tex [hunk 7]
```diff
@@ -108,7 +110,27 @@ \section*{Solution of Euler's equation}
...
+\subsection*{Notation}
+\cite{Celledoni2007} uses a dimensionless formulation where $\norm\vm = 1$, and absolute values for $I_{jh}$ and $\gD_j$.  
+We prefer to use a dimensionful formulation where $\norm\vm = G$, and to avoid absolute values.  Thus we define:
+\begin{align*}
+I_{jh} &\DefineAs I_j - I_h &\gD_j &\DefineAs G^2 - 2 T I_j &B_{jh} &\DefineAs \sqrt{±\frac{I_j \gD_h}{I_{jh}}} \\
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:118]: Done.

File: documentation/Celledoni.tex [hunk 8]
```diff
@@ -108,7 +110,27 @@ \section*{Solution of Euler's equation}
...
+\end{align*}
+With these definitions, $I_{jh} ≥ 0$ if and only if $j ≥ h$, and we will prove later that $\gD_1 ≥ 0$, $\gD_3 ≤ 0$, and that $\gD_2$ can have either sign.
+The sign under the radical in the definition of $B_{jh}$ is $+$ if $h = 1$ and $j ≥ h$, or $h = 3$ and $j < h$; it is $-$ otherwise (note that we never use
+$B_{j2}$ in the analysis below).  At this point it is also useful
+to observe that:
+\[
+B_{31}^2 + B_{13}^2 = \frac{\gD_1 I_3 - \gD_3 I_1}{I_{31}} = G^2
+\]
+
+Physically, $I_{jh}$ has the dimension of an inertial momentum $\squareBrackets{L^2 M}$.  $G$ has the dimension of an angular
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:129]: Done.

File: documentation/Celledoni.tex [hunk 9]
```diff
@@ -108,7 +110,27 @@ \section*{Solution of Euler's equation}
...
+momentum $\squareBrackets{L^2 M T^{-1} A}$.  $T$ has the dimension of an energy $\squareBrackets{L^2 M T^{-2}}$.
+$\gD_j$ has the same dimension as $G^2$.  $B_{jh}$ has the same dimension as 
+$\sqrt{\gD_h}$, i.e., the same dimension as $G$.  $\gl_1$ and $\gl_3$ have the
+same dimension as the quotient $\frac{G}{I_j}$, i.e., $\squareBrackets{T^{-1} A}$ which is appropriate for their usage.
 \subsection*{Case (i)}
 Case (i) of the solution of Euler's equation in section 2.2 of \cite{Celledoni2007} is:
 \[
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:136]: Done.

[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:136]: Done.

File: documentation/Celledoni.tex [hunk 10]
```diff
@@ -180,47 +202,48 @@ \subsection*{Case (iii)}
...
-= -\gs' \gs'' \sqrt{\frac{I_3 \gD_1}{I_{31}} \frac{I_{13}}{I_1 \gD_3}} \frac{I_2 - I_3}{I_2 I_3} \\
-&= -\gs' \gs'' \sqrt{\frac{\gD_1}{I_1 I_3 \gD_3}} \frac{I_2 - I_3}{I_2}
-= \gs' \gs'' \sqrt{\frac{\gD_1}{I_1 I_3 \gD_3}} \pa{\frac{I_3}{I_2} - 1}
+\gl &= -\gs' \gs'' G \frac{B_{31}}{B_{13}} \frac{I_2 - I_3}{I_2 I_3}
+= -\gs' \gs'' G \sqrt{\frac{I_3 \gD_1}{I_{31}} \frac{I_{13}}{I_1 \gD_3}} \frac{I_2 - I_3}{I_2 I_3} \\
+&= -\gs' \gs'' G \sqrt{-\frac{\gD_1}{I_1 I_3 \gD_3}} \frac{I_2 - I_3}{I_2}
+= \gs' \gs'' G \sqrt{-\frac{\gD_1}{I_1 I_3 \gD_3}} \pa{\frac{I_3}{I_2} - 1}
 \end{align*}
-Now note that in case (iii) we have $2 T I_2 = 1$ thus $1/I_2 = 2 T$.  $\gl$ can be rewritten as:
+Now note that in case (iii) we have $2 T I_2 = G^2$ thus $1/I_2 = \frac{2 T}{G^2}$.  $\gl$ can be rewritten as:
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:232]: Done.

File: documentation/Celledoni.tex [hunk 11]
```diff
@@ -180,47 +202,48 @@ \subsection*{Case (iii)}
...
 \[
-\gl = \gs' \gs'' \sqrt{\frac{\gD_1}{I_1 I_3 \gD_3}} \pa{2 T I_3 - 1} = \gs' \gs'' \sqrt{\frac{\gD_1 \gD_3}{I_1 I_3}}
+\gl = \gs' \gs'' G \sqrt{-\frac{\gD_1}{I_1 I_3 \gD_3}} \pa{\frac{2 T I_3}{G^2} - 1} = \gs' \gs'' \frac{1}{G} \sqrt{-\frac{\gD_1 \gD_3}{I_1 I_3}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:234]: Done.

File: documentation/Celledoni.tex [hunk 12]
```diff
@@ -272,25 +295,18 @@ \subsubsection*{Case (iii)}
...
-\subsubsection*{The quantity $I_{jh}$}
-It is simpler and more efficient to avoid absolute values, so we define:
-\[
-I_{jh} \DefineAs I_j - I_h
-\]
-This is the same quantity as in \cite{Celledoni2007} when $j \geq h$ but it has the opposite sign (it is negative) when $j < h$
+describe in this section the modifications we make to these formul{\ae} in our implementation.
 
 \subsubsection*{The quantity $\gD_j$}
 We notice that the computation of $\gD_j$ may entail cancellations, so we go back to the definition of $\norm{\vm}$ and of the kinetic energy:
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:312]: Done.

[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:312]: Sure, but let's keep the word "entails", it's cool.

File: documentation/Celledoni.tex [hunk 13]
```diff
@@ -300,10 +316,6 @@ \subsubsection*{The quantity $\gD_j$}
 2 T &= \frac{m_1^2}{I_1} + \frac{m_2^2}{I_2} + \frac{m_3^2}{I_3}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:316]: I am assuming that the radians multiply *T*.  Please comment if not.

File: documentation/Celledoni.tex [hunk 14]
```diff
@@ -300,10 +316,6 @@ \subsubsection*{The quantity $\gD_j$}
...
 \end{dcases}
 \]
-and we define a dimensionful $\gD_j$ without absolute values:
-\[
-\gD_j \DefineAs G^2 - 2 T I_j
-\]
 When, for instance, $j = 2$, this yields:
 \begin{align*}
 \gD_2 &= m_1^2 \pa{1 - \frac{I_2}{I_1}} + m_3^2 \pa{1 - \frac{I_2}{I_3}} \\
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:321]: *Ah! c’est une belle hypothèse; ça explique beaucoup de choses.*

File: documentation/Celledoni.tex [hunk 15]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\arrow{r}{\mathscr Q_{t_0}} & {\mathscr B^s}
+\end{tikzcd}
 \]
-With this adjustment we obtain the dimensionful formul{\ae}:
+
+\cite{Celledoni2007} derive the following expression for $\TimeDerivative{\gy}\of{t}$ (which they write slightly differently):
 \begin{align*}
-\gn &= -\InverseHyperbolicTangent\of{\frac{m_{02}}{G}} \\
-\gl_2 &= \frac{1}{G} \sqrt{\frac{\gD_1 \gD_3}{I_1 I_3}}
+\TimeDerivative{\gy}\of{t} &= \frac{2 T}{G} + \frac{\gD_2}{G I_2}\pa{\frac{1}{1 + \frac{I_{12}I_{23}G^2}{I_2^2 \gD_1 \gD_3}m_2^2}}\\
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:383]: After *T*?

File: documentation/Celledoni.tex [hunk 16]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+&= \frac{2 T}{G} + \frac{\gD_2}{G I_2}\pa{\frac{1}{1 - \frac{G^2}{B_{21}^2 B_{23}^2}m_2^2}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:384]: After *T*?

File: documentation/Celledoni.tex [hunk 17]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
 \end{align*}
-It is easy to verify that the argument of the $\InverseHyperbolicTangent$ is dimensionless and that $\gl_2$ has the same dimension as the quotient 
-$\frac{\gD_j}{G I_j}$, which has the same dimension as $\frac{G}{I_j}$, namely, $T^{-1} A$.
+
+\subsection*{Case (i)}
+In case (i) we have $m_2 = -B_{21} \JacobiSN\of{\gl t - \gn, k}$ and the above expression becomes:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T}{G} + \frac{\gD_2}{G I_2}\pa{\frac{1}{1 - \frac{G^2}{B_{23}^2}\JacobiSN^2\of{\gl t - \gn, k}}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:390]: After *T*?

File: documentation/Celledoni.tex [hunk 18]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+This expression can be integrated using formula 110.04 of \cite{ByrdFriedman1954} with $\ga = G/B_{23}$ to yield:
+\[
+\gy\of{t} = \frac{2 T}{G}t + \frac{\gD_2}{\gl G I_2}\EllipticPi\of{\JacobiAmplitude\of{\gl t - \gn, k}, \frac{G^2}{B_{23}^2}, k}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:394]: After *T*?

File: documentation/Celledoni.tex [hunk 19]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+Note that this differs from the formula given by \cite{Celledoni2007} in the value of $n$.
+
+\subsection*{Case (ii)}
+In case (ii) we have $m_2 = -B_{23} \JacobiSN\of{\gl t - \gn, k^{-1}}$ and a computation similar to the one above gives:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T}{G} + \frac{\gD_2}{G I_2}\pa{\frac{1}{1 - \frac{G^2}{B_{21}^2}\JacobiSN^2\of{\gl t - \gn, k^{-1}}}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:401]: After *T*?

File: documentation/Celledoni.tex [hunk 20]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+and:
+\[
+\gy\of{t} = \frac{2 T}{G}t + \frac{\gD_2}{\gl G I_2}\EllipticPi\of{\JacobiAmplitude\of{\gl t - \gn, k^{-1}}, \frac{G^2}{B_{21}^2}, k^{-1}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:405]: After *T*?

File: documentation/Celledoni.tex [hunk 21]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+
+\subsection*{Case (iii)}
+In case (ii) we have $m_2 = G \HyperbolicTangent\of{\gl t - \gn}$ and:
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:409]: Done.

File: documentation/Celledoni.tex [hunk 22]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T}{G} + \frac{\gD_2}{G I_2}\pa{\frac{1}{1 - \frac{G^4}{B_{21}^2 B_{23}^2} \HyperbolicTangent^2\of{\gl t - \gn}}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:411]: After *T*?

File: documentation/Celledoni.tex [hunk 23]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+which can be integrated using:
+\[
+\int{}\frac{1}{1 - n \HyperbolicTangent^2\of{\gl t - \gn}}\diffd{t} = 
+\frac{t}{1 - n} + \frac{\sqrt{n}}{\pa{1 - n}\gl} \InverseHyperbolicTangent\of{\sqrt{n} \HyperbolicTangent\of{\gl t - \gn}}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:416]: Done.

File: documentation/Celledoni.tex [hunk 24]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\[
+\begin{dcases}
+p_1 &= \frac{p_3 m_1 + p_0 m_2}{G + m_3} \\
+p_2 &= \frac{p_3 m_2 - p_0 m_1}{G + m_3} \\
+p_0^2 + p_3^2 &= \frac{G + m_3}{2 G}
+\end{dcases}
+\]
+and the angle $\gy\of{t}$ for the rotation $\mathscr Y_t$ around $\VectorSymbol{e_3}$:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T + G m_3 / I_3}{G + m_3} + 4 G \frac{p_3 \TimeDerivative{p_0} - p_0 \TimeDerivative{p_3}}{G + m_3}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:443]: After *T*?

File: documentation/Celledoni.tex [hunk 25]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+can be $0$.
+
+One should note at this point that \cite{Celledoni2007} make an error when copying the definition of $f_1\of{u}$ from formula 361.54 of 
+\cite{ByrdFriedman1954}.  This error is corrected in \cite{Celledoni2008}, but the sign of $f\of{s}$ in the definition of $\gy\of{t}$ is
+still incorrect and should read:
+\[
+f\of{s} \DefineAs -B_{31} \frac{B_{13}}{B_{21}} \InverseTrigonometricTangent\of{\frac{B_{21}}{B_{13}} \JacobiSD\of{\gl s - \gn, k}}
+\]
+
+\subsection*{An alternative quaternionic solution}
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:455]: Done.

File: documentation/Celledoni.tex (hunk 1 above)
Comment (pleroy) [reply documentation/Celledoni.tex:516]: After *T*?

File: documentation/Celledoni.tex (hunk 2 above)
Comment (pleroy) [reply documentation/Celledoni.tex:529]: After *T*?

File: documentation/Celledoni.tex (hunk 4 above)
Comment (pleroy) [reply documentation/Celledoni.tex:535]: I don't think so, I think you meant around line 560.

File: documentation/Celledoni.tex (hunk 3 above)
Comment (pleroy) [reply documentation/Celledoni.tex:536]: Same.

File: documentation/Celledoni.tex [hunk 26]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+Unfortunately, it is not suitable for a practical implementation because it has an essential singularity when $\TimeDerivative{\vm} = \VectorSymbol{0}$:
+while the condition $\TimeDerivative{\vm} = \VectorSymbol{0}$ is physically a constant of motion, $\TimeDerivative{\vm}$ itself is not.  This means that
+any inaccuracies in numerical computations of $\TimeDerivative{\vm}$ (cancellations, underflow) may cause it to switch from $\VectorSymbol{0}$ to
+non-$\VectorSymbol{0}$ and back.
+
+When $\TimeDerivative{\vm} = \VectorSymbol{0}$ at $t = 0$, the motion may be computed at all times assuming a constant $\vm$, so the singularity may be
+eliminated.  But when $\TimeDerivative{\vm} \neq \VectorSymbol{0}$ at $t = 0$ and it later becomes $\VectorSymbol{0}$, there is no way to find a basis at $t$ that 
+continuously corresponds to the one at $t = 0$.
+
+In conclusion, we tried to use this approach but had to abandon it because of the impossibility of handling the singularity.
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:430]: Added the text that you suggested.  Not adding figures, just a reference to figure 1.

File: documentation/Celledoni.tex [hunk 27]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+\]
+and the angle $\gy\of{t}$ for the rotation $\mathscr Y_t$ around $\VectorSymbol{e_3}$:
+\[
+\TimeDerivative{\gy}\of{t} = \frac{2 T + G m_3 / I_3}{G + m_3} + 4 G \frac{p_3 \TimeDerivative{p_0} - p_0 \TimeDerivative{p_3}}{G + m_3}
+\]
+They note that these formul{\ae} are only applicable if $m_3 \neq 0$ but go on applying them to case (i) where $m_3 = B_{31} \JacobiCN\of{\gl t - \gn, k}$
+can be $0$.
+
+One should note at this point that \cite{Celledoni2007} make an error when copying the definition of $f_1\of{u}$ from formula 361.54 of 
+\cite{ByrdFriedman1954}.  This error is corrected in \cite{Celledoni2008}, but the sign of $f\of{s}$ in the definition of $\gy\of{t}$ is
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:449]: I don't understand what you mean.  This is exactly the 𝜓(𝑡) defined by equation (17) of CFSZ07.  (Modulo the fact that the formulæ are homogeneous).

File: documentation/Celledoni.tex [hunk 28]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
...
+the denominator of the quaternionic coordinates $G + m_3$ can be safely computed.  The key insight here is that the coordinate where the Jacobi funtion
+$\JacobiDN$ appears does not change sign, and is therefore a better choice for rotating $\vm$ to $\VectorSymbol{e_i}$.
+
+Observing that, in case (i), $m_1 = \gs B_{13} \JacobiDN\of{\gl t - \gn, k}$, we will construct a rotation $\mathscr S$ to make $m_1$ positive and then
+a rotation $\mathscr P_t$ that maps $\vm$ onto $\VectorSymbol{e_1}$.  Similarly, in case (iii) the function $\HyperbolicSecant$ appears in the
+expressions of $m_1$ and $m_3$ and is always positive, so we will construct $\mathscr S$ to make both $m_1$ and $m_3$ positive and choose $\mathscr P_t$
+to map $\vm$ onto $\VectorSymbol{e_1}$ or $\VectorSymbol{e_3}$.  In the rest of this section we detail the calculations used to compute $\mathscr S$,
+$\mathscr P_t$ and $\gy\of{t}$.
+
+With the introduction of $\mathscr S$, we are effectively introducing a new base $\mathscr B^p$ for the ``preferred'' principal axes of the body, and
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:469]: Done.

File: documentation/Celledoni.tex (hunk 27 above)
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:449]: Done.

File: documentation/Celledoni.tex [hunk 29]
```diff
@@ -340,24 +352,232 @@ \subsubsection*{The elliptic modulus}
 =\frac{\gD_1 I_{32} + \gD_3 I_{21}}{\gD_1 I_{32}}
 =\frac{\gD_2 I_{31}}{\gD_1 I_{32}}
 \]
-Note that in both cases we have $m_c \geq 0$.
-
-\subsubsection*{The dimensionful equations}
-$I_{jh}$ has the physical dimension of an inertial momentum $L^2 M$.  $G$ has the dimension of an angular momentum $L^2 M T^{-1} A$.  $\gD_j$ has 
-the same dimension as $G^2$.  $B_{jh}$ has the same dimension as $\sqrt{\gD_h}$, i.e., the same dimension as $G$.  $\gl_1$ and $\gl_3$ have the
-same dimension as the quotient $\frac{G}{I_j}$, i.e., $T^{-1} A$ which is appropriate for their usage.
+In both cases we have $m_c \geq 0$.
```
[the remark this answers is not in the document]
Comment (pleroy) [reply documentation/Celledoni.tex:356]: Added a blurb, I hope that it makes some sense.

File: documentation/Celledoni.tex (hunk 4 above)
Comment (pleroy) [reply documentation/Celledoni.tex:535]: Also, I noticed that the second parameter of am(*u*, *k*) was missing in a few places.

**Comment by pleroy** [comment]
retest this please

**Pull request merged**